hawkbit-security-core/src/main/java/org/eclipse/hawkbit/security/SecurityContextTenantAware.java

```diff
@@ -59,44 +72,62 @@ public String getCurrentUsername() {
     }
 
     @Override
-    public <T> T runAsTenant(final String tenant, final TenantRunner<T> callable) {
+    public <T> T runAsTenant(final String tenant, final TenantRunner<T> tenantRunner) {
+        final SecurityContext originalContext = SecurityContextHolder.getContext();
```

Comment:
Could you please extract the wrapping context change/rollback to its own method e.g. runInContext(SecurityContext, TenantRunner), thanks!